### PR TITLE
Seamless Email Confirmation

### PR DESCRIPTION
**Pull request by bkoval**
## Links
* https://wikia-inc.atlassian.net/browse/SOC-3573

## Description

* /confirm-email?token=<token> as a route for seamless (no need to reauthenticate there) route for confirming Email
* For logged in User, clicking a link in their Email results in their email being automatically confirmed and they end up at their user page on Community wiki
* Anons need to login first (will get redirected if they're not logged in).
* If anything goes wrong with the email confirmation, user gets redirected to the main page of the Community wiki with a banner notification describing possible errors
* Hapi uses Cookie forwarding to authenticate the user in Mediawiki by access_token

## Reviewers

@Wikia/social-frontend @Grunny @garthwebb Could you take a look at this code and analyze it? Thank you!

### CODE REVIEW

File: server/app/facets/auth/confirm-email.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+/**
+ * @param {Hapi.Request} request
+ * @param {*} reply
+ * @returns {void}
+ */
+export default function get(request, reply) {
+	if (!request.auth.isAuthenticated) {
+		return reply.redirect(`http://${request.info.hostname}/signin?redirect=${getFullRequestUrl(request)}`);
```
Comment (garthwebb) [inline server/app/facets/auth/confirm-email.js]: Does the return value of getFullRequestUrl(request) get escaped anywhere?  It looks like `request.url` return s node's URL.parse() where `path` includes the query string.

Comment (Grunny) [inline server/app/facets/auth/confirm-email.js]: Yeah, the redirect param should be at least URL encoded.

Comment (bkoval) [inline server/app/facets/auth/confirm-email.js]: 👍 Right, will do!

File: server/app/facets/auth/confirm-email.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+/**
+ * @param {Hapi.Request} request
+ * @param {*} reply
+ * @returns {void}
+ */
+export default function get(request, reply) {
+	if (!request.auth.isAuthenticated) {
+		return reply.redirect(
+			`http://${request.info.hostname}/signin?redirect=${encodeURIComponent(getFullRequestUrl(request)})`
```
Comment (Grunny) [inline server/app/facets/auth/confirm-email.js]: `${encodeURIComponent(getFullRequestUrl(request)})` should be `${encodeURIComponent(getFullRequestUrl(request))}`, right?

**Pull request merged**